elements/rh-secondary-nav/rh-secondary-nav.ts

```diff
@@ -0,0 +1,389 @@
+import { RhSecondaryNavDropdown, SecondaryNavDropdownChangeEvent } from './rh-secondary-nav-dropdown.js';
+
+import { tabletLandscapeBreakpoint } from '../../lib/tokens.js';
+import { MatchMediaController } from '../../lib/MatchMediaController.js';
+
+import styles from './rh-secondary-nav.css';
+
+/**
+ * Red Hat Secondary Nav
+ *
+ * @summary A non primary navigation bar.
+ *
+ * @slot base           - Slot to override all shadow dom contents inside nav
+ * @slot logo           - Logo added to the main nav bar, expects a `<a> | <a><svg/></a> | <a><img/></a>`
+ * @slot nav            - Navigation list added to the main nav bar, expects a `<ul>`
+ * @slot cta            - Nav bar level CTA, expects a `<pfe-cta>
+ *
+ * @csspart base        - {HTMLNavElement} container, <nav> element
+ * @csspart container   - {HTMLElement} css grid container, <div> element
+ * @csspart cta         - {HTMLElement} container, <div> element
+ *
+ * @fires { SecondaryNavOverlayEvent }  - Fires when an dropdown is opened or closed in desktop view or when
+ *                                        the mobile menu button is toggled in mobile view.
+ */
+@customElement('rh-secondary-nav') @pfelement()
+export class RhSecondaryNav extends LitElement {
+  static readonly version = '{{version}}';
+
+  static readonly styles = [styles];
+
+  #logger = new Logger(this);
+
+  /**
+   * executes this.shadowRoot.querySelector('rh-secondary-nav-overlay')
+   */
+  @query('rh-secondary-nav-overlay') _overlay: RhSecondaryNavOverlay | undefined;
+
+  /**
+   * executes this.shadowRoot.querySelector('#container')
+   */
+  @query('#container') _container: HTMLElement | undefined;
+
+  /**
+   * executes this.shadowRoot. querySelctor('nav');
+   */
+  @query('nav') _nav: HTMLElement | undefined;
+
+  /**
+   * executes this.shadowRoot.querySelector('button')
+   */
+  @query('button') _mobileMenuButton: HTMLButtonElement | undefined;
+
+
+  /**
+   * `isMobile` property is true when viewport `(min-width: ${tabletLandscapeBreakpoint})`.
+   * Property is observed for changes, and its value is updated using matchMediaControler
+   * when viewport changes at breakpoint or first load of the component.
+   */
+  @observed
+  @property({ reflect: false }) _isMobile = false;
+
+  /**
+   * Checks if passed in element is a RhSecondaryNavDropdown
+   * @param element:
+   * @returns {boolean}
+   */
+  static isDropdown(element: Element|null): element is RhSecondaryNavDropdown {
+    return element instanceof RhSecondaryNavDropdown;
+  }
+
+  /**
+   * Updates isMobile property with matchMedia when viewport changes
+   */
+  protected matchMedia = new MatchMediaController(this, `(min-width: ${tabletLandscapeBreakpoint})`, {
+    onChange: ({ matches }) => {
+      this._isMobile = !matches;
+    }
+  });
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.addEventListener('change', this._dropdownChangeHandler as EventListener);
```

Comment:
in this case i recommend sticking with lib's typing and do an `instanceof` check inside the handler. if a user slots a `select` or `input` element and it changes...